pkg/storage/storagewrappers/caching_test.go

```diff
@@ -54,37 +50,9 @@ func TestReadAuthorizationModel(t *testing.T) {
 	gotModel, err = cachingBackend.ReadAuthorizationModel(ctx, storeID, model.Id)
 	require.NoError(t, err)
 	require.Equal(t, model, gotModel)
-}
-
-func TestSingleFlightFindLatestAuthorizationModelID(t *testing.T) {
-	const numGoroutines = 2
-
-	mockController := gomock.NewController(t)
-	defer mockController.Finish()
-	mockDatastore := mockstorage.NewMockOpenFGADatastore(mockController)
-	expectedModelID := "expectedId"
-	mockDatastore.EXPECT().FindLatestAuthorizationModelID(gomock.Any(), gomock.Any()).DoAndReturn(func(ctx context.Context, storeID string) (string, error) {
-		time.Sleep(1 * time.Second)
-		return expectedModelID, nil
-	}).Times(1)
-	mockDatastore.EXPECT().Close().Times(1)
 
-	cachingBackend := NewCachedOpenFGADatastore(mockDatastore, 5)
-	defer cachingBackend.Close()
-
-	var wg errgroup.Group
-	for i := 0; i < numGoroutines; i++ {
-		wg.Go(func() error {
-			id, err := cachingBackend.FindLatestAuthorizationModelID(context.Background(), "id")
-			if err != nil {
-				return err
-			}
-			if id != expectedModelID {
-				return fmt.Errorf("expected model ID %s, actual %s", expectedModelID, id)
-			}
-			return nil
-		})
-	}
-	err := wg.Wait()
+	// ensure find latest authorization model will get hte latest model
```

Comment:
This assertion isn't enough. we should change

```go
memoryBackend := memory.New()
cachingBackend := NewCachedOpenFGADatastore(memoryBackend, 5)
```

to 

```go
mockController := gomock.NewController(t)
defer mockController.Finish()
mockDatastore := mocks.NewMockOpenFGADatastore(mockController)
mockDatastore.EXPECT().FindLatestAuthorizationModel(gomock.Any(), gomock.Any()).Times(1)
cachingBackend := NewCachedOpenFGADatastore(mockDatastore, 5)
```

note the `Times(1)` :)

Reply:
FindLatestAuthorizationModel is not serving from cache though

Reply:
But this is a unit test for the cache decoration

Reply:
updated